### DISCUSSION TITLE
Navigating level up in library displays JSON

### DISCUSSION THREAD

**Discussion by Lubos-V** [rern/rAudio#1199] [Bug Reports]
Hi, have observed this over the last couple of days (was on UPDATE, now updated to latest release, stilll there).

How to replicate:
- open library (NAS)
- navigate to the album of interest
- add to queue
- try to navigate level up
- instead of the respective screen, the corresponding JSON is displayed
- using the breadcrumb navigation on top works fine

**Comment by Lubos-V** [comment]
this is still happening - interestingly enough, only on an iPhone (Safari browser). any idea what might be causing the behavior and how to fix it?

**Reply by Lubos-V** [reply]
@rern even after recent update, this is still happening on Safari on iPhone. anything that could be done about this?

**Reply by kw-sudo** [reply]
Did you use ‚album view (1) or navigated through NAS source (2)? In both ways I can‘t replicate it on my iPhone (13, iOS 16.5.1(c)).

![image](https://github.com/rern/rAudio/assets/104136223/2402609d-6652-4764-82a1-3b0376a354b3)

PS: to complete, same on iPad

Oliver

**Reply by Lubos-V** [reply]
hi @master-kw I am using the NAS source navigation (up-to-date iPhone XR).

i start here - navigating down the tree to the album level:
![image0](https://github.com/rern/rAudio/assets/120736629/7b4f518b-c54a-47ac-a7b2-fb4cc0ad9f5c)

using the "back arrow" displays this:
![image1](https://github.com/rern/rAudio/assets/120736629/e5b32b83-af73-47d6-9ed5-dc8cd3ceb0bc)

after closing this view, it returns back to the album level. using the breadcrumb (in this case clicking on "A") brings me back to the desired higher level:
![image2](https://github.com/rern/rAudio/assets/120736629/2c625ddb-2b32-47b6-89a7-7514d77f47b8)

it has been like this for months. in the meantime I already re-flashed / made a new install, but this is happening consistently. on my PCs, no issues...

**Reply by Lubos-V** [reply]
@master-kw I can also confirm this is happenning on an older Samsung device of my wife.

when playback is running, the JSON data shown contains all the details related to the currently running track.

**Reply by kw-sudo** [reply]
Do you only do r/audio addon updates or additionally a complete archlinuxarm with _pacman -Syu_?

Since I do the latter regularly, at least another json-error has settled for me.

Oliver

**Reply by Lubos-V** [reply]
@master-kw only the r/audio addon updates. just finished the fupp update - when we are finished with listening, will reboot the system and check, mayube it has helped... immediately after update (without reboot) i don't see any difference...

**Reply by Lubos-V** [reply]
@master-kw even after full update, no change in the behaviour...

**Comment by surfninja2** [comment]
i wonder if the JASON parser is out of date.  I'm using an old RP:
rAudio 20230630
5.10.44-1-ARCH armv6l
Raspberry PiB Rev 2
BCM2835 455 MiB
ARM1176 @ 900 MHz

and continue to get JSON errors during playback.  I've been ssh into the RP and running 'mpc stop' to turn off the player.  not very 'web friendly'.  I tried to compare the libraries between my RP2 (running rAudio great) and this old RP.  They seem different.

I'm not familiar with **pacman -Syu**.

Can I just ssh into my RP and run it from the command line?

**Reply by kw-sudo** [reply]
Sure, you can. If you want to you can first sync your databases by using _pacman -Syy_, the update by -Syu

Oliver

**Reply by Lubos-V** [reply]
have done that @surfninja2 @master-kw - the problem is still there but as I said, only on the iPhone browser. strange, ain't it?

**Reply by rern** [reply]
@Lubos-V,

Please try:
```sh
sed -i '/#button-lib-back/ a\	z-index: 115;' /srv/http/assets/css/main.css
cacheBust
```

**Reply by Lubos-V** [reply]
@rern not sure what **cacheBust** is

executed the sed command and rebooted, but to no avail. still behaves the same.

**Reply by kan-ibal** [reply]
> been ssh into the RP and running 'mpc stop' to turn off the player. not very 'web friendly'. 

But why? MPC is designed to work remotely and is 'web friendly'. There is no need to 'ssh'.

**Reply by surfninja2** [reply]
i forgot about the --host=xxx.xxx.xx.xxx option.

**Comment by surfninja2** [comment]
> Sure, you can. If you want to you can first sync your databases by using _pacman -Syy_, the update by -Syu
> 
> Oliver
**pacman -Syy** returns:

**pacman -Syy**
_:: Synchronizing package databases...
 +R                      7.0 KiB  19.5 KiB/s 00:00 [######################] 100%
 core.db failed to download
error: failed retrieving file 'core.db' from ca.us.mirror.archlinuxarm.org : The requested URL returned error: 404
 error: failed retrieving file 'core.db' from dk.mirror.archlinuxarm.org : The requested URL returned error: 404_

with all my mirrors.  I went down the [rabbit hole](https://bbs.archlinux.org/viewtopic.php?id=266340) looking at that error and haven't come back to the JSON issue.

it seems like a package is out of date.  my rp2 and rp0 2w are fine.

I updated /etc/packman.d/mirrorlist and packman found core.db but running **packman -Syu** returned:
_:: Starting full system upgrade...
error: could not open file /var/lib/pacman/sync/core.db: Unrecognized archive format
error: could not open file /var/lib/pacman/sync/extra.db: Unrecognized archive format
error: could not open file /var/lib/pacman/sync/alarm.db: Unrecognized archive format
:: Replace hwids with +R/hwdata? [Y/n]_ 

i might have hosed up the system. 30+ years of unix/linux developement but arch is kind of new to me.  at least **vi** is available.

ran uname and got this:
#  uname -a
Linux rAudio-1 5.10.44-1-ARCH #1 SMP Fri Jun 18 02:04:20 UTC 2021 armv6l GNU/Linux

**Reply by rern** [reply]
Arch Linux Arm - [End of life for v5/v6](https://archlinuxarm.org/forum/viewtopic.php?f=3&t=15721)

**Comment by surfninja2** [comment]
for fun, I spun up a RP zero W from the downloads (rAudio-RPi0-1-20230521.img.xz) and get the same JASON error.  Then I spun up a RP zero 2 W (rAudio-64bit-20230521.img.xz) and it works fine.

based on my crude testing, looks like the RP1 and zero (ver1) have the same JSON errors.  I'll continue to use ssh and 'mpc stop' but i'll check back for any updates.

I can test either one if it helps rern or anyone else looking for a solution.

I hope I'm not thread cr*ping.  if so, sorry about that!

**Reply by Lubos-V** [reply]
@surfninja2 @rern just to make sure - I am seeing this behaviour on RPi 3B with the newest code (and have been seeing it for quite some time now)
